Subject: build-sys: do not depend on check_rundir for executables
From: Michael Tokarev <mjt@tls.msk.ru>
To: linux-raid@vger.kernel.org

The problem is that the link lines will always be executed
even if the executables are up to date.

If anything, this check_rundir should be a dependency of
install target, or some other "phony" target like this,
since check_rundir is phony by itself.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>

--- a/Makefile
+++ b/Makefile
@@ -172,7 +172,7 @@ everything-test: all mdadm.static swap_super test_stripe \
 # mdadm.uclibc and mdassemble.uclibc don't work on x86-64
 # mdadm.tcc doesn't work..
 
-mdadm : check_rundir $(OBJS)
+mdadm : $(OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o mdadm $(OBJS) $(LDLIBS)
 
 mdadm.static : $(OBJS) $(STATICOBJS)
@@ -195,7 +195,7 @@ mdmon.O2 : $(MON_SRCS) $(INCL) mdmon.h
 	$(CC) -o mdmon.O2 $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(MON_LDFLAGS) -DHAVE_STDINT_H -O2 -D_FORTIFY_SOURCE=2 $(MON_SRCS)
 
 # use '-z now' to guarantee no dynamic linker interactions with the monitor thread
-mdmon : check_rundir $(MON_OBJS)
+mdmon : $(MON_OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(MON_LDFLAGS) -Wl,-z,now -o mdmon $(MON_OBJS) $(LDLIBS)
 msg.o: msg.c msg.h
 
